superset/views/core.py

```diff
@@ -188,7 +188,10 @@ def check_datasource_perms(
     except SupersetException as e:
         raise SupersetSecurityException(str(e))
 
-    viz_obj = get_viz(  # type: ignore
+    if datasource_type is None:
```

Comment:
I dislike the `get_datasource_info(...)` method from  a typing perspective. The `datasource_type` will be non-None (the method raises otherwise) and thus I think `assert datasource_type` is suffice from a type checking standpoint.

Reply:
But it doesn't hurt to give a more user-friendly error message, would it?

Reply:
`get_datasource_info(...)` only raises on `datasource_id` is `None`, but not on `datasource_type`.